src/GitVersionCore/GitRepoInformation/GitRepoMetadata.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System.Collections.Generic;
+
+namespace GitVersion.GitRepoInformation
+{
+    public class GitRepoMetadata
```

Comment:
I am unsure at the moment to be honest. I don't want to re-implement git data structures, this really is information about the commit graph with a bunch of pre-calculated stuff. Lets see where it goes then fix naming next